### PR TITLE
Check for berserk before eating item

### PR DESCRIPTION
**Pull request by Karthik99999**
https://github.com/smogon/pokemon-showdown/projects/3#card-52059203

Here's a video of it working twice during the same turn in a double battle: https://imgur.com/a/AFVlZdY

The solution does hardcode in the eatItem function, but I don't believe there is another way to do it, since eatItem is called during onUpdate by the berries.

### CODE REVIEW

**Comment by DaWoblefet** [comment]
A bit hardcodey but this is actually a VGC-relevant mechanic. The proper way is likely identifying every place where Berries can be eaten, which would be a pretty large refactor. I'll let Marty give what he thinks.

File: data/abilities.ts [hunk 1]
```diff
@@ -300,14 +300,24 @@ export const Abilities: {[abilityid: string]: AbilityData} = {
 		num: 224,
 	},
 	berserk: {
+		onDamage(damage, target) {
+			target.abilityData.checkedBerserk = false;
+		},
 		onAfterMoveSecondary(target, source, move) {
-			if (!source || source === target || !target.hp || !move.totalDamage) return;
```
[the remark this answers is not in the document]
Comment (Karthik99999) [reply data/abilities.ts:304]: yeah looks like I can, will add it

**Comment by Marty-D** [comment]
I'm okay with this solution, but you'd also need to ensure this only applies to Berries that restore HP. I'd expect Petaya and Jaboca, for example, to activate even if Berserk would activate.

**Review by Marty-D — APPROVED** [review]
Great, this is better than hardcoding in eatItem, thanks! 👍

**Comment by DaWoblefet** [comment]
Thanks a bunch!

**Pull request merged**